src/util/memory.rs

```diff
@@ -105,6 +105,48 @@ pub fn set(start: Address, val: u8, len: usize) {
     }
 }
 
+/// Set a range of memory to the given pattern. Similar to C++'s std::fill.
+pub fn set_pattern(start: Address, pattern: usize, len: usize) {
```

Comment:
I suggest using a fixed-size integer type (or a type trait) for the pattern.  At the call site, we give the pattern as a hexadecimal literal, but here we write memory at the granularity of `usize` which is platform-dependent.  For example, if `pattern` is `0xdeadbeef`, and the platform is x86_64, then the memory will be filled with `EF BE AD DE 00 00 00 00 EF BE AD DE 00 00 00 00 EF BE ...`, and I think we don't want the `00 00 00 00` in between.

```suggestion
pub fn set_pattern(start: Address, pattern: u32, num_bytes: usize) {
```

Alternatively

```suggestion
trait UnsignedIntButNotUsize: num_traits::PrimInt + num_traits::Unsigned + num_traits::ToBytes {}
impl UnsignedIntButNotUsize for u8 {}
impl UnsignedIntButNotUsize for u16 {}
impl UnsignedIntButNotUsize for u32 {}
impl UnsignedIntButNotUsize for u64 {}

pub fn set_pattern<T: UnsignedIntButNotUsize>(start: Address, pattern: T, num_bytes: usize) {
```